### ISSUE TITLE
Unable to install local tgz file after upgraded to node 10

### ISSUE THREAD

**Issue by a9udn9u** [yarnpkg/yarn#5761]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**
Install local tgz file either hang at 'Resolving packages' or exit after 'Fetching packages'

**If the current behavior is a bug, please provide the steps to reproduce.**

I've created this [test package](https://www.dropbox.com/s/yn7zcqbewk5edte/yarn-tgz.tar.gz?dl=1). Just run `yarn add file:../b.tgz` in package `a`. b.tgz is packed by `yarn pack`.

**What is the expected behavior?**
tgz file gets unpacked and installed.

**Please mention your node.js, yarn and operating system version.**
yarn: 1.6.0
nodejs: 10.0.0
OS: macOS 10.12.6

**More details.**
The problem started right after I upgrade to node 10, before it worked fine.

Sometimes yarn hang at 'Resolving packages'
```
yarn add v1.6.0
(node:1191) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
verbose 0.544 Checking for configuration file "/private/tmp/yarn/a/.npmrc".
verbose 0.544 Checking for configuration file "/Users/quangan/.npmrc".
verbose 0.545 Found configuration file "/Users/quangan/.npmrc".
verbose 0.545 Checking for configuration file "/usr/local/etc/npmrc".
verbose 0.546 Checking for configuration file "/private/tmp/yarn/a/.npmrc".
verbose 0.546 Checking for configuration file "/private/tmp/yarn/.npmrc".
verbose 0.546 Checking for configuration file "/private/tmp/.npmrc".
verbose 0.546 Checking for configuration file "/private/.npmrc".
verbose 0.549 Checking for configuration file "/private/tmp/yarn/a/.yarnrc".
verbose 0.549 Checking for configuration file "/Users/quangan/.yarnrc".
verbose 0.549 Found configuration file "/Users/quangan/.yarnrc".
verbose 0.549 Checking for configuration file "/usr/local/etc/yarnrc".
verbose 0.55 Checking for configuration file "/private/tmp/yarn/a/.yarnrc".
verbose 0.55 Checking for configuration file "/private/tmp/yarn/.yarnrc".
verbose 0.55 Checking for configuration file "/private/tmp/.yarnrc".
verbose 0.55 Checking for configuration file "/private/.yarnrc".
verbose 0.552 current time: 2018-04-30T16:59:20.229Z
info No lockfile found.
[1/4] 🔍  Resolving packages...
⢀ file:../b.tgz
```

Sometimes it dies after 'Fetching packages'
```
yarn add v1.6.0
(node:5003) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
verbose 0.261 Checking for configuration file "/tmp/app/.npmrc".
verbose 0.261 Checking for configuration file "/Users/quangan/.npmrc".
verbose 0.261 Checking for configuration file "/usr/local/etc/npmrc".
verbose 0.261 Checking for configuration file "/tmp/app/.npmrc".
verbose 0.262 Checking for configuration file "/tmp/.npmrc".
verbose 0.263 Checking for configuration file "/tmp/app/.yarnrc".
verbose 0.263 Checking for configuration file "/Users/quangan/.yarnrc".
verbose 0.263 Found configuration file "/Users/quangan/.yarnrc".
verbose 0.263 Checking for configuration file "/usr/local/etc/yarnrc".
verbose 0.263 Checking for configuration file "/tmp/app/.yarnrc".
verbose 0.263 Checking for configuration file "/tmp/.yarnrc".
verbose 0.264 Found configuration file "/Users/quangan/.yarnrc".
verbose 0.264 Checking for configuration file "/Users/.yarnrc".
verbose 0.265 current time: 2018-04-27T05:11:53.672Z
[1/4] 🔍  Resolving packages...
verbose 0.401 Performing "GET" request to "https://registry.yarnpkg.com/uuid".
verbose 0.545 Request "https://registry.yarnpkg.com/uuid" finished with status code 200.
[2/4] 🚚  Fetching packages...
[#################################################################################################################################################################################################################################] 834/835
```

There is no yarn-error.log file.

**Comment by imsnif** [comment]
Hey @a9udn9u - thanks for bringing this up. This issue is addressed by https://github.com/yarnpkg/yarn/pull/5769

Hoping to have it merged soon.

**Comment by theandrewlane** [comment]
@imsnif - I have verified your fix resolves this issue on MacOS High Sierra 10.13.3 w/Node 10.1.0 👍
